lib/fs.js

```diff
@@ -1191,7 +1191,7 @@ function writeFile(path, data, options, callback) {
   function writeFd(fd, isUserFd) {
     const buffer = isArrayBufferView(data) ?
       data : Buffer.from('' + data, options.encoding || 'utf8');
-    const position = /a/.test(flag) ? null : 0;
```

Comment:
For clarity I would use braces:

```suggestion
  let position = (/a/.test(flag) || isUserFd) ? null : 0;
```